modules/images/webp-uploads/helper.php

```diff
@@ -240,3 +240,19 @@ function webp_uploads_get_attachment_sources( $attachment_id, $size = 'thumbnail
 	// Return an empty array if no sources found.
 	return array();
 }
+
+/**
+ * Verify if the request if for the frontend context.
+ * The is_feed() function is to check if the query is for feeds (rdf, rss, rss2, atom).
+ *
+ * @since n.e.x.t
+ *
+ * @return bool
+ */
+function webp_uploads_is_frontend_context() {
```

Comment:
Curios on why not just use `!is_feed()` additionally we might want to find a different name since this might be confused with the existing `is_front_page()` function that already exists but just test the front page.

Reply:
> Curios on why not just use `!is_feed()`

There are other non-frontend contexts, like REST API. And plugins can also do whatever they want outside of a frontend context, so we should really use the checks for the hooks which are the only reliable way.

> additionally we might want to find a different name since this might be confused with the existing `is_front_page()` function that already exists but just test the front page

I'm less concerned about confusion with `is_front_page()`, however I agree the name here could be refined, because it doesn't check just the context. It also checks whether we're in the part of the HTML frontend that is shown to the user, a.k.a. not the `head`.

How about
```suggestion
function webp_uploads_in_frontend_body() {
```